File: w02/11053.cpp
#include <iostream>

using namespace std;

// 1) 
int n;
int arr[1000001] = {0,};
int dp[1000001] = {1,};
int res = 0;

int main () {
    cin >> n;
    for (int i=0; i<n; i++) {
        cin >> arr[i];
    }

    for (int i=0; i<n; ++i) {
        dp[i] = 1;
        for (int j=0; j<i; ++j) {
            if (arr[i] > arr[j]) {
                dp[i] = max(dp[i], dp[j]+1);
            }
        }
    }

    for (int i = 0; i < n; ++i) {
        res = max(res, dp[i]);
    }
    cout<<res;

    return 0;
}

File: w02/ivk/11053.py
n = int(input())
arr = list(map(int, input().split()))

l = [1] * n

for i in range(1, n):
    for j in range(i):
        if arr[j] < arr[i] and l[i] < l[j] + 1:
            l[i] = l[j] + 1

print(max(l))




